### PR TITLE
Le Kidney Beans

### PR DESCRIPTION
**Pull request by LadyFowls**
- rscadd: "Adds the ability to remove appendix and other organs in character setup."
  - rscadd: "Adds two kidneys! Splits the work up between the two organs."

Also adds the ability for a mechanical liver and kidneys,appendix

### CODE REVIEW

**Comment by LadyFowls** [comment]
!review

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 1]
```diff
@@ -110,6 +110,8 @@
 							I.mechassist()
 						if ("mechanical")
 							I.robotize()
+						if ("removed")
+							qdel(I)
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_helpers.dm]: I think you need to call the organ's removed proc, unless its Destroy() does that already?

File: code/modules/organs/internal/kidneys.dm [hunk 2]
```diff
@@ -25,21 +25,37 @@
...
 			if(status & ORGAN_DEAD)
-				owner.adjustToxLoss(1)
\ No newline at end of file
+				owner.adjustToxLoss(0.5)
+
+/obj/item/organ/internal/kidneys/attackby(var/obj/item/I, var/mob/user)
+	..()
+	if(istype(I, /obj/item/hemostat))
+		if(src.organ_tag == BP_KIDNEYS)
+			to_chat(usr, "You reattach the kidneys for placement on the right side.")
```
Comment (MarinaGryphon) [inline code/modules/organs/internal/kidneys.dm]:
```suggestion
			to_chat(usr, "You prepare the kidney for placement on the right side.")
```

File: code/modules/organs/internal/kidneys.dm [hunk 3]
```diff
@@ -25,21 +25,37 @@
...
+			organ_tag = BP_RKIDNEYS
+			name = "right kidney"
+		else if(organ_tag == BP_RKIDNEYS)
+			organ_tag = BP_KIDNEYS
+			to_chat(usr, "You reattach the kidneys for placement on the left side.")
```
Comment (MarinaGryphon) [inline code/modules/organs/internal/kidneys.dm]:
```suggestion
			to_chat(usr, "You prepare the kidney for placement on the left side.")
```

File: code/modules/organs/internal/kidneys.dm [hunk 4]
```diff
@@ -25,21 +25,37 @@
...
+			name = "left kidney"
+/obj/item/organ/internal/kidneys/right
+	name = "right kidney"
+	robotic_name = "prosthetic right kidneys"
```
Comment (MarinaGryphon) [inline code/modules/organs/internal/kidneys.dm]:
```suggestion
	robotic_name = "prosthetic right kidney"
```

File: code/modules/mob/living/carbon/human/human_helpers.dm (hunk 1 above)
Comment (LadyFowls) [reply code/modules/mob/living/carbon/human/human_helpers.dm]: Destroy() Does most of what removed does

**Comment by KasparoVy** [comment]
2 eyes, and the ability to choose a missing one for authentic piracy

**Comment by NonQueueingMatt** [comment]
I don't like the way this has been done at all. It's pretty snowflakey, makes the kidneys overall stronger and will be a pain in the ass when someone adds something that affects the kidneys in other code.

File: code/__defines/mobs.dm [hunk 5]
```diff
@@ -74,7 +74,8 @@
 #define BP_LUNGS    "lungs"
 #define BP_BRAIN    "brain"
 #define BP_LIVER    "liver"
-#define BP_KIDNEYS  "kidneys"
+#define BP_KIDNEYS  "left kidney"
```
Comment (Geevies) [inline code/__defines/mobs.dm]: why does it say "kidneys" if it refers to a single one, the left one?

File: code/modules/organs/internal/kidneys.dm [hunk 6]
```diff
@@ -1,10 +1,10 @@
 /obj/item/organ/internal/kidneys
```
Comment (Geevies) [inline code/modules/organs/internal/kidneys.dm]: kidneys but only refers to left

File: code/modules/research/designs/mechfab/prosthetics/internal.dm [hunk 7]
```diff
@@ -24,6 +24,11 @@
 	id = "robotic_kidneys"
 	build_path = /obj/item/organ/internal/kidneys
 
+/datum/design/item/mechfab/prosthetic/internal/kidneys
+	name = "Prosthetic Right Kidneys"
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/prosthetics/internal.dm]: this has the same path as the previous one, which will probably override it. not to mention that the previous one still just says "Kidneys", not "Left Kidney"

File: code/modules/client/preference_setup/general/03_body.dm [hunk 8]
```diff
@@ -637,6 +656,11 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 				pref.organ_data[organ] = "assisted"
 			if("Mechanical")
 				pref.organ_data[organ] = "mechanical"
+			if("Removed")
+				if((organ in list(BP_HEART, BP_LIVER)) || (organ == BP_LKIDNEY && pref.organ_data[BP_RKIDNEY] == "removed") || (organ == BP_RKIDNEY && pref.organ_data[BP_LKIDNEY]))
+					to_chat(user, "<span class='notice'>You cannot start without a heart.</span>")
```
Comment (KasparoVy) [inline code/modules/client/preference_setup/general/03_body.dm]: `to_chat(user, "<span class='notice'>You cannot start without a [organ].</span>")`

**Comment by KasparoVy** [comment]
Thanks!

File: code/modules/organs/internal/kidneys.dm (hunk 6 above)
Comment (MarinaGryphon) [reply code/modules/organs/internal/kidneys.dm]: This is what we do with left/right arms, it's normal practice.

**Comment by skull132** [comment]
welp I made a mistake and merged this while on auto-pilot.

ANYWAYS. Ye no, this method of handling duplicating organs is not the best. The `/2` is fine. But what's not largely fine is stuff like requiring two organ tags per one set of duplicating organ. Ideally, they'd be listed under one tag. This way, life() doesn't need too much modification if you want to duplicate whatever organ, for example.

**Pull request merged**